lib/api/objectPut.js

```diff
@@ -78,18 +81,29 @@ function _storeIt(bucketName, objectKey, objMD, authInfo, canonicalID,
     // 0 or > 0 - existing object with content-length 0 or greater than 0
     const prevContentLen = objMD && objMD['content-length'] !== undefined ?
         objMD['content-length'] : null;
+
+    const backendInfoObj = locationConstraintCheck(request, null, bucket, log);
+    if (backendInfoObj.err) {
+        return process.nextTick(() => {
```

Comment:
As discussed with Lauren, this makes the synchronous check work asynchronously for consistency in the file :)